### PR TITLE
 packages luarocks updater as pyproject package

### PR DESCRIPTION
**Pull request by teto**
## Description of changes

Extracted from https://github.com/NixOS/nixpkgs/pull/342884

This made it easier to set as a python dependency of the neovim updater + pyproject has the advantage that we can configure the linting in it for more coherence between contributors.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/lu/luarocks-packages-updater/package.nix [hunk 1]
```diff
@@ -16,52 +12,45 @@ let
...
-  format = "other";
+  src = lib.cleanSource ./.;
 
   nativeBuildInputs = [
     makeWrapper
-    python3Packages.wrapPython
+    python3Packages.setuptools
   ];
+
   propagatedBuildInputs = [
```
Comment (GaetanLepage) [inline pkgs/by-name/lu/luarocks-packages-updater/package.nix:31]:
```suggestion
  dependencies = [
```

File: pkgs/by-name/lu/luarocks-packages-updater/package.nix [hunk 2]
```diff
@@ -16,52 +12,45 @@ let
...
+  inherit pname version;
+  pyproject = true;
 
-  format = "other";
+  src = lib.cleanSource ./.;
 
   nativeBuildInputs = [
     makeWrapper
-    python3Packages.wrapPython
+    python3Packages.setuptools
```
Comment (GaetanLepage) [inline pkgs/by-name/lu/luarocks-packages-updater/package.nix:28]: Technically, this should go in `build-system`  but it is more cosmetic than useful so not a blocker.

**Review by GaetanLepage — COMMENTED** [review]
Overall LGTM, have you been able to test it ?

File: pkgs/by-name/lu/luarocks-packages-updater/package.nix (hunk 2 above)
Comment (teto) [reply pkgs/by-name/lu/luarocks-packages-updater/package.nix:27]: I've replaced with build-system

File: pkgs/by-name/lu/luarocks-packages-updater/package.nix (hunk 1 above)
Comment (teto) [reply pkgs/by-name/lu/luarocks-packages-updater/package.nix]: ty for reminding me about the update

**Comment by teto** [comment]
> Overall LGTM, have you been able to test it ?

Ofc I've tested it :D if you merge, I thinkg all commits can be squashed

File: pkgs/by-name/lu/luarocks-packages-updater/package.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/by-name/lu/luarocks-packages-updater/package.nix:27]: Thanks ! You have gotten rid of `makeWrapper`. Was that on purpose ?

Comment (teto) [reply pkgs/by-name/lu/luarocks-packages-updater/package.nix:27]: yes it seems to work without, the buildPythonPackage brings it automatically I think

Comment (GaetanLepage) [reply pkgs/by-name/lu/luarocks-packages-updater/package.nix:27]: Ok cool !

**Review by GaetanLepage — APPROVED** [review]
Thanks for this !

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 343584` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>luarocks-packages-updater</li>
    <li>luarocks-packages-updater.dist</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Pull request merged**